src/lightning/pytorch/loops/fetchers.py

```diff
@@ -42,24 +43,28 @@ def combined_loader(self) -> CombinedLoader:
 
     def setup(self, combined_loader: CombinedLoader) -> None:
         self._combined_loader = combined_loader
+        self.length = sized_len(combined_loader)
+        self.done = self.length == 0
```

Comment:
should we add a test case for this?

Reply:
https://github.com/Lightning-AI/lightning/pull/18376/files#r1303403909 covers it